js/modules/k6/experimental/tracing/propagator.go

```diff
@@ -52,18 +106,59 @@ const (
 	// Its value is zero, which is described in the Jaeger documentation as:
 	// "0 value is valid and means “root span” (when not ignored)"
 	JaegerRootSpanID = "0"
+
+	// JaegerSampledTraceFlag is the trace-flag value for an unsampled trace.
+	JaegerSampledTraceFlag = "0"
+
+	// JaegerUnsampledTraceFlag is the trace-flag value for a sampled trace.
+	JaegerUnsampledTraceFlag = "1"
 )
 
 // JaegerPropagator is a Propagator for the Jaeger trace context header
-type JaegerPropagator struct{}
+type JaegerPropagator struct {
+	// Sampler is used to determine whether or not a trace should be sampled.
+	Sampler
+}
+
+// NewJaegerPropagator returns a new JaegerPropagator with the given sampler.
+//
+// Note that we allocate the propagator on the heap to ensure we conform
+// to the Sampler interface, as the [Sampler.SetSamplingRate]
+// method has a pointer receiver.
+func NewJaegerPropagator(s Sampler) *JaegerPropagator {
+	return &JaegerPropagator{
+		Sampler: s,
+	}
+}
 
 // Propagate returns a header with a random trace ID in the Jaeger format
 func (p *JaegerPropagator) Propagate(traceID string) (http.Header, error) {
 	spanID := randHexString(8)
-	// flags set to 1 means the span is sampled
-	flags := "1"
+	flags := p.SamplingFlag()
 
 	return http.Header{
 		JaegerHeaderName: {traceID + ":" + spanID + ":" + JaegerRootSpanID + ":" + flags},
 	}, nil
 }
+
+// SamplingFlag returns the trace sample flag for the trace,
+// based on the current sampling rate.
+//
+// It uses under the `Sampled` method under the hood to
+// set the flag to either W3CSampledTraceFlag or W3CUnsampledTraceFlag,
+// with a percent of chance based on the propagator's sampling
+// rate.
+func (p *JaegerPropagator) SamplingFlag() string {
+	if p.Sampled() {
+		return JaegerSampledTraceFlag
+	}
+
+	return JaegerUnsampledTraceFlag
+}
```

Comment:
I kind of feel like this can be refactored to be a function that based on `Sampled` returns one of two strings that are given to it.

```golang
   flags := ternaryOperator(p.Sampled(), JaegerSampledTraceFlag, JaegerUnsampledTraceFlag)
```

Given that his is basically what happens in both implementations. I also kind of don't see what else could happen ... maybe there is only *1* option :shrug: 

Maybe the header isn't sent at all :shrug: 

Both of those can be handled with the "ternaryOperator" 

I am not really happy with how my code works as well but it does drop > 20 lines of code and a interface method so seems better to me